Cray compile failure

## Description
Compiling with the Cray compiler fails like this:

In file included from /home/users/david.davies/cylc-run/mo-bundle-414/share/mo-bundle/orca-jedi/src/orca-jedi/state/State.cc:42:
In file included from /home/users/david.davies/cylc-run/mo-bundle-414/share/mo-bundle/orca-jedi/src/orca-jedi/utilities/IOUtils.h:13:
/home/users/david.davies/cylc-run/mo-bundle-414/share/mo-bundle/orca-jedi/src/orca-jedi/nemo_io/ReadServer.h:33:15: error: explicitly defaulted move assignment operator is implicitly deleted [-Werror,-Wdefaulted-function-deleted]
  ReadServer &operator=(ReadServer &&) = default;
              ^
/home/users/david.davies/cylc-run/mo-bundle-414/share/mo-bundle/orca-jedi/src/orca-jedi/nemo_io/ReadServer.h:61:16: note: move assignment operator of 'ReadServer' is implicitly deleted because field 'mpiroot' is of const-qualified type 'const size_t' (aka 'const unsigned long')
  const size_t mpiroot = 0;
               ^
/home/users/david.davies/cylc-run/mo-bundle-414/share/mo-bundle/orca-jedi/src/orca-jedi/nemo_io/ReadServer.h:34:15: error: explicitly defaulted copy assignment operator is implicitly deleted [-Werror,-Wdefaulted-function-deleted]
  ReadServer &operator=(const ReadServer &) = default;
              ^
/home/users/david.davies/cylc-run/mo-bundle-414/share/mo-bundle/orca-jedi/src/orca-jedi/nemo_io/ReadServer.h:61:16: note: copy assignment operator of 'ReadServer' is implicitly deleted because field 'mpiroot' is of const-qualified type 'const size_t' (aka 'const unsigned long')
  const size_t mpiroot = 0;


## Instructions to reproduce the bug
Build with Cray compiler.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,12 +58,14 @@ endif()
 ################################################################################
 # Project-wide compiler options
 
-message( "   CMAKE_CXX_COMPILER_ID : ${CMAKE_CXX_COMPILER_ID}" )
+message(STATUS "   CMAKE_CXX_COMPILER_ID : ${CMAKE_CXX_COMPILER_ID}" )
 if(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
 endif()
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+  if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "CrayClang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+  endif()
 endif()
 if(CMAKE_Fortran_COMPILER_ID MATCHES "GNU|Clang")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -Werror")

